src/source-nfq.c

```diff
@@ -536,6 +536,9 @@ static int NFQCallBack(struct nfq_q_handle *qh, struct nfgenmsg *nfmsg,
     if (ret == -1) {
 #ifdef COUNTERS
         NFQQueueVars *q = NFQGetQueue(ntv->nfq_index);
+        if (!q) {
+            return -1;
```

Comment:
The `Packet` object obtained from `PacketGetFromQueueOrAlloc()` must be returned on this (and the other new error return that you added).

Check the other error cases -- I think `TmqhOutputPacketpool` will do it or maybe even calling `NFQReleasePacket`.

Reply:
```
        if (q) {
            q->errs++;
            q->pkts++;
            q->bytes += GET_PKT_LEN(p);
        }
```
is it okay solution? I'm not sure what you want from me

Reply:
Yes, this seems ok for this change and the second change in the PR.